docs/lang/articles/contribution/dev_install.md

```diff
@@ -4,318 +4,467 @@ sidebar_position: 2
-  previous builds. This allows a fresh build without any cache from the previous
-  builds. Note that to uninstall the Taichi package from your Python
-  environment, please use `pip uninstall taichi`.
+The `develop` command serves the developers' needs better because edits to the Python files take effect immediately without the need to rerun the command. A rerun is needed only if you have modified the project's C extension or compiled files. See the [Development Mode](https://setuptools.pypa.io/en/stable/userguide/development_mode.html) for more information.
 
-## Conda
+:::
 
-To avoid directly installing Taichi's dependencies into your existing
-Python environment, we have provided a pre-defined `conda` environment.
-You can find the instructions [here](https://github.com/taichi-dev/taichi/blob/master/conda/README.md).
+</TabItem>
 
-:::note
-This step only helps you setup the development environment,
-you would still need to run `python3 setup.py develop` to re-build
-Taichi.
-:::
+<TabItem value="windows">
 
-## Docker
+1. Run powershell as administrator and `Set-ExecutionPolicy RemoteSigned`.
+2. Run [this script](https://github.com/taichi-dev/taichi_assets/releases/download/windows_scripts/install_deps.ps1) to install all necessary dependencies:
 
-For those who prefer to use Docker, we also provide a Dockerfile which
-helps setup the Taichi development environment with CUDA support based
-on Ubuntu docker image.
+   ```shell
+   .\install_deps.ps1
+   ```
+3. Reboot your computer. 
+4. Run powershell as administrator and run [this script](https://github.com/taichi-dev/taichi_assets/releases/download/windows_scripts/build_taichi.ps1) to build Taichi from souce: 
 
-:::note
-In order to follow the instructions in this section, please make sure
-you have the [Docker Desktop (or Engine for
-Linux)](https://www.docker.com/products/docker-desktop) installed and
-set up properly.
-:::
+   ```shell
+   .\build_taichi.ps1
+   ```
 
-### Build the Docker image
+</TabItem>
 
-From within the root directory of the taichi Git repository, execute
-`docker build -t taichi:latest .` to build a Docker image based off the
-local master branch tagged with _latest_. Since this builds the image
-from source, please expect up to 40 mins build time if you don't have
-cached Docker image layers.
+</Tabs>
 
-:::note
 
-In order to save the time on building Docker images, you could always
-visit our [Docker Hub
-repository](https://hub.docker.com/r/taichidev/taichi) and pull the
-versions of pre-built images you would like to use.
+## Install optional dependencies
 
-For example, to pull a image built from release v0.6.17, run
-`docker pull taichidev/taichi:v0.6.17`
-:::
+CUDA is Nvidia's answer to high-performance computing. Taichi has implemented a backend based on CUDA 10.0.0+. Vulkan is a next-generation, cross-platform API, open standard for 3D graphics and computing. Taichi has added a Vulkan backend as of v0.8.0. 
```

Comment:
Nvidia -> NVIDIA?

> CUDA is Nvidia's answer to high-performance computing.

Better to use an introduction from an official source.

Reply:
Should be "NVIDA". Thanks for spotting this. We cannot copy and paste an official CUDA introduction, but I'll add a link to the wikipedia source.